File: library_management/library_management/report/test_abc_123/test_abc_123.py
'''import frappe

def get_columns():
    return [
        {
            "label": "ID",
            "fieldname": "name",
            "fieldtype": "Data",
            "width": 140
        },
        {
            "label": "Title",
            "fieldname": "title",
            "fieldtype": "Data",
            "width": 200
        },
        {
            "label": "Customer",
            "fieldname": "customer",
            "fieldtype": "Link",
            "options": "Customer",
            "width": 200
        },
        {
            "label": "Status",
            "fieldname": "status",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": "Grand Total",
            "fieldname": "grand_total",
            "fieldtype": "Currency",
            "width": 140
        }
    ]

def get_data(filters=None):
    if not filters:
        filters = {}

    return frappe.get_all(
        "Sales Invoice",
        fields=[
            "name",
            "title",
            "customer",
            "status",
            "grand_total"
        ],
        filters=filters,
        limit_page_length=50
    )

def execute(filters=None):
    columns = get_columns()
    data = get_data(filters)
    return columns, data
'''

File: library_management/library_management/report/test_abc_123/test_abc_123.js
frappe.query_reports["Test ABC 123"] = {
    formatter: function(value, row, column, data, default_formatter) {
        let formatted = default_formatter(value, row, column, data);

        // Right-align Grand Total
        if (column.fieldname === "grand_total") {
            formatted = `<div style="text-align:right; padding:2px;">${value}</div>`;
        }

        // Padding for Customer
        if (column.fieldname === "customer") {
            formatted = `<div style="padding:2px;">${value}</div>`;
        }

        // Padding for Status
        if (column.fieldname === "status") {
            formatted = `<div style="padding:2px;">${value}</div>`;
        }

        // Padding for Invoice ID (name)
        if (column.fieldname === "name") {
            formatted = `<div style="padding:2px;">${value}</div>`;
        }

        return formatted;
    }
};
